src/components/elements/Map-mapbox/MapControls/PolygonHandler.tsx

```diff
@@ -0,0 +1,179 @@
+        break;
+      case "kml":
+        uploadPromise = fetchPostV2TerrafundUploadKmlProject({ body: newRequest });
+        break;
+      default:
+        break;
+    }
+
+    try {
+      if (uploadPromise) {
+        const response = await uploadPromise;
+        if (response instanceof Blob) {
+          displayNotification(t("File uploaded successfully"), "success", t("Success!"));
+          const blob = response;
+          const url = window.URL.createObjectURL(blob);
+          const a = document.createElement("a");
+          const getFormattedDate = () => {
+            const date = new Date();
+            const year = date.getFullYear();
+            const month = String(date.getMonth() + 1).padStart(2, "0");
+            const day = String(date.getDate()).padStart(2, "0");
+            return `${year}-${month}-${day}`;
+          };
+
+          const currentDate = getFormattedDate();
+          a.href = url;
+          a.download = `polygon-check-results-${currentDate}.csv`;
+          a.click();
+          window.URL.revokeObjectURL(url);
+        }
+      }
+    } catch (error) {
+      if (error && typeof error === "object" && "message" in error) {
+        let errorMessage = error.message as string;
+        const parsedMessage = JSON.parse(errorMessage);
+        if (parsedMessage && typeof parsedMessage === "object" && "message" in parsedMessage) {
+          errorMessage = parsedMessage.message;
+        }
+        displayNotification(t("Error uploading file"), "error", errorMessage);
+      } else {
+        displayNotification(t("Error uploadig file"), "error", t("An unknown error occurred"));
+      }
+    }
+    hideLoader();
+    closeModal();
+    reloadSiteData?.();
+  };
+
+  const getFileType = (file: UploadedFile) => {
+    const fileType = file?.file_name.split(".").pop()?.toLowerCase();
+    if (fileType === "geojson") return "geojson";
+    if (fileType === "zip") return "shapefile";
+    if (fileType === "kml") return "kml";
+    return null;
+  };
+
+  const openFormModalHandlerConfirmUpload = (type: string) => {
+    openModal(
+      <ModalConfirm
+        title={t(`Confirm Polygon ${type}`)}
+        content={t(
+          `${
+            type === "Creation" ? "Creating" : "Uploading"
+          } a new polygon will overwrite the existing geometry. Proceed?`
+        )}
+        onClose={() => closeModal()}
+        onConfirm={() => {
+          type === "Creation" ? setIsUserDrawingEnabled(true) : openFormModalHandlerAddPolygon();
+        }}
+      />
+    );
+  };
+
+  const openFormModalHandlerAddPolygon = () => {
+    openModal(
+      <ModalAdd
+        title="Add Polygons"
+        descriptionInput={`Drag and drop a GeoJSON, Shapefile, or KML.`}
+        descriptionList={
+          <div className="mt-9 flex">
+            <Text variant="text-12-bold">TerraMatch upload limits:&nbsp;</Text>
+            <Text variant="text-12-light">50 MB per upload</Text>
```

Comment:
make sure all free text is translated in this component. I see these lines and `content` and `primaryButtonText` below at least needing an update.

Reply:
its ready